hw6/task1/library.js

```diff
@@ -0,0 +1,194 @@
+                                if(Array.isArray(myArguments[i])){
+                                    if(myArguments[i][i-1].constructor.name === "Track" ){                                                                                             
+                                        element.traks = element.traks.concat(myArguments[i]);
+                                    }
+                                    if(myArguments[i][i-1].constructor.name === "Album"){
+                                        element.albums = element.albums.concat(myArguments[i]);
+                                    }                                 
+                                }
+                                if(myArguments[i].constructor.name === "Track"){
+                                        element.traks.push(myArguments[i]);
+                                }
+                                if(myArguments[i].constructor.name === "Album"){
+                                    element.albums.push(myArguments[i]);
+                                }
+                                                         
+                            }
+                        } 
+                    });
+                }                           
+            };         
+    };        
+
+// DELETE METHOD 
+    delete(value){
+                    //TRACK
+        if(value.constructor.name === "Track"){
+           let index = this.Track.indexOf(value);
+           this.Track.splice(index,1);               
+        };
+                    //ALBUM
+        if(value.constructor.name === "Album"){
+            let index = this.Album.indexOf(value);
+            this.Album.splice(index,1);               
+         };
+                    //ARTIST
+        if(value.constructor.name === "Artist"){
+            let index = this.Artist.indexOf(value);
+            this.Artist.splice(index,1);               
+         };         
+    };
+
+// UPDATE METHOD
+    update(myElement, argument, value ){
+                    //TRACK
+        if(myElement.constructor.name === "Track"){
+            this.Track.forEach(element => {
+                if(element === myElement){
+                    element[`${argument}`] = value;
+                }
+            });            
+        }
+                    //ALBUM
+        if(myElement.constructor.name === "Album"){
+            this.Album.forEach(element => {
+                if(element === myElement){
+                    element[`${argument}`] = value;
+                }
+            });            
+        }
+        if(myElement.constructor.name === "Artist"){
+            this.Artist.forEach(element => {
+                if(element === myElement){
+                    element[`${argument}`] = value;
+                }
+            });            
+        }        
+    }
+    
+//PLAY METHOD
+    play(){ 
+        const EventEmitter = require('events');
```

Comment:
all imports should be declared at the start of file, no matter where we use this variable